### ISSUE TITLE
Can't install textract

### ISSUE THREAD

**Issue by HarshitDaftary** [deanmalmgren/textract#243]
Facing below error:

Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-mux5kues/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-14vl23ml/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-install-mux5kues/pocketsphinx/

**Comment by Cathay2017** [comment]
I meet the same issue to install textract in  Ubuntu 16.04.  Any solution?
  Running setup.py bdist_wheel for pocketsphinx ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5w3wbj_p/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-x10i3_sq --python-tag cp35:
  running bdist_wheel
  running build_ext
  building 'sphinxbase._ad' extension
  swigging swig/sphinxbase/ad.i to swig/sphinxbase/ad_wrap.c
  swig -python -modern -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -Ideps/sphinxbase/swig -outdir sphinxbase -o swig/sphinxbase/ad_wrap.c swig/sphinxbase/ad.i
  creating build
  creating build/temp.linux-x86_64-3.5
  creating build/temp.linux-x86_64-3.5/swig
  creating build/temp.linux-x86_64-3.5/swig/sphinxbase
  creating build/temp.linux-x86_64-3.5/deps
  creating build/temp.linux-x86_64-3.5/deps/sphinxbase
  creating build/temp.linux-x86_64-3.5/deps/sphinxbase/src
  creating build/temp.linux-x86_64-3.5/deps/sphinxbase/src/libsphinxad
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPHINXBASE_EXPORTS -DPOCKETSPHINX_EXPORTS -DSPHINX_DLL -DHAVE_CONFIG_H -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -I/usr/include/python3.5m -c swig/sphinxbase/ad_wrap.c -o build/temp.linux-x86_64-3.5/swig/sphinxbase/ad_wrap.o -Wno-unused-label -Wno-strict-prototypes -Wno-parentheses -Wno-unused-but-set-variable -Wno-unused-variable -Wno-unused-result -Wno-sign-compare -Wno-misleading-indentation
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPHINXBASE_EXPORTS -DPOCKETSPHINX_EXPORTS -DSPHINX_DLL -DHAVE_CONFIG_H -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -I/usr/include/python3.5m -c deps/sphinxbase/src/libsphinxad/ad_pulse.c -o build/temp.linux-x86_64-3.5/deps/sphinxbase/src/libsphinxad/ad_pulse.o -Wno-unused-label -Wno-strict-prototypes -Wno-parentheses -Wno-unused-but-set-variable -Wno-unused-variable -Wno-unused-result -Wno-sign-compare -Wno-misleading-indentation
  deps/sphinxbase/src/libsphinxad/ad_pulse.c:44:30: fatal error: pulse/pulseaudio.h: No such file or directory
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for pocketsphinx
  Running setup.py clean for pocketsphinx
Successfully built textract python-pptx docx2txt EbookLib
Failed to build pocketsphinx
Installing collected packages: argcomplete, chardet, python-pptx, docx2txt, beautifulsoup4, xlrd, EbookLib, SpeechRecognition, pocketsphinx, textract
  Found existing installation: chardet 3.0.4
    Uninstalling chardet-3.0.4:
      Successfully uninstalled chardet-3.0.4
  Found existing installation: docx2txt 0.7
    Uninstalling docx2txt-0.7:
      Successfully uninstalled docx2txt-0.7
  Found existing installation: beautifulsoup4 4.4.1
    Uninstalling beautifulsoup4-4.4.1:
      Successfully uninstalled beautifulsoup4-4.4.1
  Found existing installation: xlrd 1.1.0
    Uninstalling xlrd-1.1.0:
      Successfully uninstalled xlrd-1.1.0
  Running setup.py install for pocketsphinx ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5w3wbj_p/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-n49ddal2/install-record.txt --single-version-externally-managed --compile:
    running install
    running build_ext
    building 'sphinxbase._ad' extension
    swigging swig/sphinxbase/ad.i to swig/sphinxbase/ad_wrap.c
    swig -python -modern -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -Ideps/sphinxbase/swig -outdir sphinxbase -o swig/sphinxbase/ad_wrap.c swig/sphinxbase/ad.i
    creating build
    creating build/temp.linux-x86_64-3.5
    creating build/temp.linux-x86_64-3.5/swig
    creating build/temp.linux-x86_64-3.5/swig/sphinxbase
    creating build/temp.linux-x86_64-3.5/deps
    creating build/temp.linux-x86_64-3.5/deps/sphinxbase
    creating build/temp.linux-x86_64-3.5/deps/sphinxbase/src
    creating build/temp.linux-x86_64-3.5/deps/sphinxbase/src/libsphinxad
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPHINXBASE_EXPORTS -DPOCKETSPHINX_EXPORTS -DSPHINX_DLL -DHAVE_CONFIG_H -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -I/usr/include/python3.5m -c swig/sphinxbase/ad_wrap.c -o build/temp.linux-x86_64-3.5/swig/sphinxbase/ad_wrap.o -Wno-unused-label -Wno-strict-prototypes -Wno-parentheses -Wno-unused-but-set-variable -Wno-unused-variable -Wno-unused-result -Wno-sign-compare -Wno-misleading-indentation
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPHINXBASE_EXPORTS -DPOCKETSPHINX_EXPORTS -DSPHINX_DLL -DHAVE_CONFIG_H -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -Ideps/sphinxbase/include/android -I/usr/include/python3.5m -c deps/sphinxbase/src/libsphinxad/ad_pulse.c -o build/temp.linux-x86_64-3.5/deps/sphinxbase/src/libsphinxad/ad_pulse.o -Wno-unused-label -Wno-strict-prototypes -Wno-parentheses -Wno-unused-but-set-variable -Wno-unused-variable -Wno-unused-result -Wno-sign-compare -Wno-misleading-indentation
    deps/sphinxbase/src/libsphinxad/ad_pulse.c:44:30: fatal error: pulse/pulseaudio.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5w3wbj_p/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-n49ddal2/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-5w3wbj_p/pocketsphinx/

**Comment by matt32106** [comment]
Just to be on the safe side, did you install the pre-requisites? 

> Ubuntu / Debian
> 
> There are two steps required to run this package on Ubuntu/Debian. First you must install some system packages using the apt-get package manager before installing textract from pypi.
> 
> apt-get install python-dev libxml2-dev libxslt1-dev antiword unrtf poppler-utils pstotext tesseract-ocr \
> flac ffmpeg lame libmad0 libsox-fmt-mp3 sox libjpeg-dev swig libpulse-dev
> pip install textract

**Comment by RomHartmann** [comment]
Here it is:
https://stackoverflow.com/questions/5178416/libxml-install-error-using-pip/5178444#5178444

```
apt-get install libpulse-dev
```

So you need
```
apt-get install python-dev libxml2-dev libxslt1-dev antiword unrtf poppler-utils pstotext tesseract-ocr flac ffmpeg lame libmad0 libsox-fmt-mp3 sox libjpeg-dev swig libpulse-dev
```

**Comment by mjspeck** [comment]
I have the same issue on MacOS:

```
Running setup.py bdist_wheel for pocketsphinx ... error
  Complete output from command /Users/matthewspeck/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-install-tmkaz6oe/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-wheel-067naz02 --python-tag cp36:
  running bdist_wheel
  running build_ext
  building 'sphinxbase._ad' extension
  swigging swig/sphinxbase/ad.i to swig/sphinxbase/ad_wrap.c
  swig -python -modern -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -I/System/Library/Frameworks/OpenAL.framework/Versions/A/Headers -Ideps/sphinxbase/include/android -Ideps/sphinxbase/swig -outdir sphinxbase -o swig/sphinxbase/ad_wrap.c swig/sphinxbase/ad.i
  unable to execute 'swig': No such file or directory
  error: command 'swig' failed with exit status 1

  ----------------------------------------
  Failed building wheel for pocketsphinx
  Running setup.py clean for pocketsphinx
Failed to build pocketsphinx
Installing collected packages: pocketsphinx, textract
  Found existing installation: pocketsphinx 0.1.15
    Uninstalling pocketsphinx-0.1.15:
      Successfully uninstalled pocketsphinx-0.1.15
  Running setup.py install for pocketsphinx ... error
    Complete output from command /Users/matthewspeck/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-install-tmkaz6oe/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-record-rlthdw2c/install-record.txt --single-version-externally-managed --compile:
    running install
    running build_ext
    building 'sphinxbase._ad' extension
    swigging swig/sphinxbase/ad.i to swig/sphinxbase/ad_wrap.c
    swig -python -modern -Ideps/sphinxbase/include -Ideps/sphinxbase/include/sphinxbase -I/System/Library/Frameworks/OpenAL.framework/Versions/A/Headers -Ideps/sphinxbase/include/android -Ideps/sphinxbase/swig -outdir sphinxbase -o swig/sphinxbase/ad_wrap.c swig/sphinxbase/ad.i
    unable to execute 'swig': No such file or directory
    error: command 'swig' failed with exit status 1

    ----------------------------------------
  Rolling back uninstall of pocketsphinx
Command "/Users/matthewspeck/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-install-tmkaz6oe/pocketsphinx/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-record-rlthdw2c/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /private/var/folders/2w/m42b2ghj3fn6t0sq7fvw47lr0000gn/T/pip-install-tmkaz6oe/pocketsphinx/
```

**Comment by idansc** [comment]
> Here it is:
> https://stackoverflow.com/questions/5178416/libxml-install-error-using-pip/5178444#5178444
> 
> ```
> apt-get install libpulse-dev
> ```
> 
> So you need
> 
> ```
> apt-get install python-dev libxml2-dev libxslt1-dev antiword unrtf poppler-utils pstotext tesseract-ocr flac ffmpeg lame libmad0 libsox-fmt-mp3 sox libjpeg-dev swig libpulse-dev
> ```

What about CentOS?

**Comment by jpweytjens** [comment]
Closed by #292. Please install the latest version of textract with

**Comment by jpweytjens** [comment]
Closed by #292. Please install the latest version of textract with

`pip install git+https://github.com/deanmalmgren/textract.git`
